server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CassandraDumbBlobStoreCache.java

```diff
@@ -1,119 +1,162 @@
-                        .flatMap(chunk -> save(blobId, chunk)).single();
-                }
-                return Mono.error(new IllegalArgumentException(
-                    String.format("Data must not biger than %s kb", cacheConfiguration.MAX_SIZE)));
-            });
+        BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream, cacheConfiguration.getSizeThreshold() + 1);
+        return Mono.fromCallable(() -> isGreaterThanThreshold(bufferedInputStream))
+            .filter(bigStream -> !bigStream)
+            .flatMap(ignored ->
+                Mono.fromCallable(() -> IOUtils.toByteArray(bufferedInputStream))
+                    .flatMap(bytes -> Mono.from(save(blobId, ByteBuffer.wrap(bytes, 0, bytes.length)))))
+            .switchIfEmpty(Mono.empty())
+            .doFinally(any -> closeInputStreamQuite(inputStream));
+
     }
 
     @Override
     public Publisher<byte[]> read(BlobId blobId) {
-        return cassandraAsyncExecutor.executeSingleRow(select.bind().setString(ID, blobId.asString()))
+        return cassandraAsyncExecutor
+            .executeSingleRow(
+                selectStatement.bind()
+                    .setString(ID, blobId.asString())
+                    .setConsistencyLevel(ConsistencyLevel.ONE))
             .map(row -> row.getBytes(DATA).array())
-            .timeout(cacheConfiguration.TIME_OUT, Mono.empty());
+            .timeout(cacheConfiguration.getTimeOut(), Mono.empty());
     }
 
     @Override
-    public Publisher<Void> delete(BlobId blobId) {
+    public Publisher<Void> remove(BlobId blobId) {
         return cassandraAsyncExecutor.executeVoid(
-            delete.bind().setString(ID, blobId.asString()));
+            deleteStatement.bind().setString(ID, blobId.asString()));
+    }
+
+    private PreparedStatement prepareDelete(Session session) {
+        return session.prepare(
+            delete().
+                from(BlobTables.DumbBlobCache.TABLE_NAME)
+                .where(eq(ID, bindMarker(ID))));
+    }
+
+    private PreparedStatement prepareSelect(Session session) {
+        return session.prepare(
+            select()
+                .from(BlobTables.DumbBlobCache.TABLE_NAME)
+                .where(eq(ID, bindMarker(ID))));
+    }
+
+    private PreparedStatement prepareInsert(Session session) {
+        return session.prepare(
+            insertInto(BlobTables.DumbBlobCache.TABLE_NAME)
+                .value(ID, bindMarker(ID))
+                .value(DATA, bindMarker(DATA))
+                .using(ttl(bindMarker(TTL)))
+        );
+    }
+
+    private boolean isGreaterThanThreshold(BufferedInputStream bufferedData) throws IOException {
+        bufferedData.mark(0);
+        bufferedData.skip(cacheConfiguration.getSizeThreshold());
+        boolean readable = bufferedData.read() != -1;
+        bufferedData.reset();
+        return readable;
+    }
+
+    private void closeInputStreamQuite(InputStream inputStream) {
```

Comment:
closeInputStreamQuite  ? Why `Quite` ?

Reply:
Because it dont throw when cannot close inputStream, i call it is

Reply:
The name should be "Quiet" then.